### DISCUSSION TITLE
Alternatives to subprocess execution

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18287] [Q&A] [answered]
Hello. Whenever I'm materializing a new asset in a run, I see a log message saying Dagster is

&gt; Launching subprocess for "my_asset".
&gt; Executing step "my_asset" in subprocess.
The time spent launching the sub-processes is non-trivial (given many assets are very simple &lt; 5 second computations)

Am I missing something? Is there a way to get the wall-time execution of a run to be faster by sharing processes or something comparable.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700846301622999?thread_ts=1700846301.622999&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
You can specify the [`in_process` executor in the launchpad](https://docs.dagster.io/concepts/ops-jobs-graphs/job-execution#default-job-executor), which means that compute steps will execute in a single process as opposed to the default multi-process executor. This can be helpful for jobs with short-running compute steps.
